### PR TITLE
gl_engine: revise target API for context managing

### PR DESCRIPTION
**Pull request by tinyjin**
Issue: #3024

### CODE REVIEW

File: src/renderer/tvgGlCanvas.cpp [hunk 1]
```diff
@@ -59,8 +59,11 @@ GlCanvas::~GlCanvas()
 }
 
 
-Result GlCanvas::target(int32_t id, uint32_t w, uint32_t h, ColorSpace cs) noexcept
+Result GlCanvas::target(void* context, int32_t id, uint32_t w, uint32_t h, ColorSpace cs) noexcept
 {
+#ifdef __EMSCRIPTEN__
+    emscripten_webgl_make_context_current((EMSCRIPTEN_WEBGL_CONTEXT_HANDLE)context);
```
Comment (RuiwenTang) [inline src/renderer/tvgGlCanvas.cpp:65]: make some check may help improve the performance:
```
if (emscripten_webgl_get_current_context() != context) emscripten_webgl_make_context_current(context);
```

Comment (hermet) [reply src/renderer/tvgGlCanvas.cpp:65]: gl call should be inside gl_engine, not canvas.

File: src/renderer/gl_engine/tvgGlRenderer.h [hunk 2]
```diff
@@ -115,6 +115,7 @@ class GlRenderer : public RenderMethod
 
     void clearDisposes();
 
+    void* mContext;
```
Comment (hermet) [inline src/renderer/gl_engine/tvgGlRenderer.h:118]: `mContext = nullptr;`

File: src/bindings/capi/thorvg_capi.h [hunk 3]
```diff
@@ -519,6 +519,7 @@ TVG_API Tvg_Canvas* tvg_glcanvas_create(void);
 * This function specifies the drawing target where the rasterization will occur. It can target
 * a specific framebuffer object (FBO) or the main surface.
 *
+* @param[in] context The GL context.
```
Comment (hermet) [inline src/bindings/capi/thorvg_capi.h:522]: `The GL context assigning to the current canvas rendering.`

**Review by hermet — COMMENTED** [review]
Probably we can apply this for webgl but It still misses the native systems.

**Review by RuiwenTang — APPROVED** [review]

**Review by hermet — COMMENTED** [review]
@tinyjin Please apply to these functions as well.

```
GlRenderer::sync()
GlRenderer::preRender()
GlRenderer::clear()
bool GlRenderer::target(int32_t id, uint32_t w, uint32_t h)
bool GlRenderer::preUpdate()
```

**Comment by hermet** [comment]
@tinyjin I updated with a subsequent change. please confirm with web test. Thanks.

**Pull request merged**